Readme.md

```diff
@@ -1,25 +1,87 @@
+The idea is that the flat files that this crate can decode could also be used as an archival format similar to era1 files, specially
+if they can be verified. 
+
+## Getting Started
 
 ### Prerequisites
-- Rust installed
-- Cargo installed
-- [protoc installed](https://grpc.io/docs/protoc-installation/)
+- [Rust (stable)](https://www.rust-lang.org/tools/install)
+- Cargo (Comes with Rust by default)
+- [protoc](https://grpc.io/docs/protoc-installation/)
 - Firehose dbin files to decode
   - An example file is provided `example0017686312.dbin`
 
-### Running
-- Run `cargo run --release` in the root directory of the project
-- The program will decode all files in the input_files directory
-  - It will verify the receipt root & transaction root matches the computed one for all blocks
+## Running
+
+### Commands
+
+The tool provides the following commands for various operations:
+
+- `stream`: Stream data continuously.
+- `decode`: Decode files from input to output.
+- `help`: Print this message or the help of the given subcommand(s).
 
 ### Options
-- `--input <path>`: Specify a directory or single file to read from (default: input_files)
-- `--output <path>`: Specify a directory to write output files to (if missing it will not write to disk)
 
-### Benchmarks
+You can use the following options with the commands for additional functionalities:
+
+- `-h, --help`: Print help information about specific command and options.
+- `-V, --version`: Print the version information of the tool.
+
+
+#### NOTICE: either streaming or reading from directory it will verify the receipt root & transaction root matches the computed one for all blocks
+
+## Usage Examples
+
+Here are some examples of how to use the commands:
+
+1. To stream data continuously from `stdin`:
+
+  ```bash
+  # simply turning on stream stdin reading
+  cargo run stream
+  
+  # or from files into stdin
+  cat example0017686312.dbin | cargo run stream
+  ```
+
+This will output decoded header records as bytes into `stdout`
+
+2. To check a folder of dbin files:
+
+```bash
+cargo run decode --input ./input_files/
+```
+
+This will store the block headers as json format in the output folder. 
+By passing `--headers-dir` a folder of assumed valid block headers can be provided to compare
+with the input flat files. Valid headers can be pulled from the [sync committee subprotocol](https://github.com/ethereum/annotated-spec/blob/master/altair/sync-protocol.md) for post-merge data.
+
+<!-- TODO: once the header_accumulator is made public, link it here -->
+**NOTICE:**For pre-merge data another approach using [header accumulators](https://github.com/ethereum/portal-network-specs/blob/8ad5bc33cb0d4485d2eab73bf2decc43e7566a8f/history-network.md#the-header-accumulator) is necessary since
+sync committees will not provide these headers.
+
+## Goals
+<!-- TODO: Any other goals I should add? -->
```

Comment:
Maybe a note about how we hope Indexers/Firehose operators will use this, e.g.:

"Our goal is to provide The Graph's Indexers the tools to trustlessly share flat files with cryptographic guarantees that the data in the flat files is part of the canonical history of the Ethereum blockchain, enabling Indexers to quickly sync all historical data and begin serving data with minimal effort."